### DISCUSSION TITLE
using api.run confuses compilers like nuitka

### DISCUSSION THREAD

**Discussion by nickger** [IfcOpenShell/IfcOpenShell#4355] [Ideas]
FYI
I am not entirely sure to what extent it can be important for others, but for me pretty much. I compile my scripts with nuitka to make them faster and transferable. I noticed that "api.run" commands which are so popular in documetation, are not correctly recognized by nuitka - it cannot understand what exactly these commands are doing and which modules are to be imported. I expect the same issue with cython and similar. Using explicit classes with "Usecase" method does not cause any issues.

**Comment by Moult** [comment]
This is a known limitation of the current architecture. There is a proposed fix (though the extents to how far we go on being defensive on typing and schema differences are not yet confirmed) here https://github.com/IfcOpenShell/IfcOpenShell/pull/3851

Unfortunately due to lack of resources we haven't yet tackled it.

**Comment by nickger** [comment]
Got it. Would love to see that ticket as a part of documentation about the overall logic of API - e.g. for me it looks contra-intuitive.
